src/Transport/AwsAuthV4.php

```diff
@@ -42,15 +43,28 @@ private function getSigningMiddleware(): callable
             : \getenv('AWS_REGION');
         $signer = new SignatureV4('es', $region);
         $credProvider = $this->getCredentialProvider();
+        $transport = $this;
 
         return Middleware::mapRequest(static function (RequestInterface $req) use (
             $signer,
-            $credProvider
+            $credProvider,
+            $transport
         ) {
-            return $signer->signRequest($req, $credProvider()->wait());
+            return $signer->signRequest($transport->sanitizeRequest($req), $credProvider()->wait());
         });
     }
 
+    private function sanitizeRequest(RequestInterface $request): RequestInterface
+    {
+        // Trailing dots are valid parts of DNS host names (see RFC 1034),
+        // but interferes with header signing where AWS expects a stripped host name.
+        if ('.' === \substr($request->getHeader('host')[0], -1)) {
+            $request = Utils::modifyRequest($request, ['set_headers' => ['host' => \rtrim($request->getHeader('host')[0], '.')]]);
```

Comment:
Based on 

https://github.com/ruflin/Elastica/blob/7e4de1eebdcbc5a4731fda61e098aaf40a8d0258/src/Transport/Guzzle.php#L207-L210

Would this work with guzzle 6? I guess the same check should be one here.

Reply:
It would work with Guzzle 6.3, but not with earlier versions as far as I can see.

Would something like

```
private function sanitizeRequest(RequestInterface $request): RequestInterface
    {
        // Trailing dots are valid parts of DNS host names (see RFC 1034),
        // but interferes with header signing where AWS expects a stripped host name.
        if ('.' === \substr($request->getHeader('host')[0], -1)) {
            $changes = ['set_headers' => ['host' => \rtrim($request->getHeader('host')[0], '.')]];
            if (\class_exists(Psr7\Utils::class)) {
                $request = Psr7\Utils::modifyRequest($request, $changes);
            } else {
                $request = Psr7\modify_request($request, $changes);
            }
        }

        return $request;
    }
```
work for you/in this case, @franmomu ? If yes, I'll commit that.

Reply:
yeah, my concern was about the existing code that apparently works for both Guzzle 6 and 7, this seems like a proper way to deal with it 👍